projects/igniteui-angular-elements/src/analyzer/printer.ts

```diff
@@ -0,0 +1,132 @@
+        importPath = path.relative(path.dirname(this.#configPath), importPath);
+        return importPath.replace(path.extname(importPath), '');
+    }
+
+    #createContentQueryLiteral(query: ContentQuery) {
+        const properties = [
+            ts.factory.createPropertyAssignment('property', ts.factory.createStringLiteral(query.property)),
+            ts.factory.createPropertyAssignment('childType', ts.factory.createIdentifier(query.childType.symbol.name))
+        ];
+
+        if (query.isQueryList) {
+            properties.push(ts.factory.createPropertyAssignment('isQueryList', ts.factory.createToken(ts.SyntaxKind.TrueKeyword)));
+        }
+
+        if (query.descendants) {
+            properties.push(ts.factory.createPropertyAssignment('descendants', ts.factory.createToken(ts.SyntaxKind.TrueKeyword)));
+        }
+
+        return ts.factory.createObjectLiteralExpression(properties);
+    }
+
+    #createMetaLiteralObject([type, meta]: readonly [ts.InterfaceType, ComponentMetadata]) {
+        const properties = [
+            ts.factory.createPropertyAssignment('component', ts.factory.createIdentifier(type.symbol.name)),
+            ts.factory.createPropertyAssignment('parents', ts.factory.createArrayLiteralExpression(meta.parents.map(x => ts.factory.createIdentifier(x.symbol.name)))),
+            ts.factory.createPropertyAssignment('contentQueries', ts.factory.createArrayLiteralExpression(meta.contentQueries.map(x => this.#createContentQueryLiteral(x)))),
+            ts.factory.createPropertyAssignment('methods', ts.factory.createArrayLiteralExpression(meta.methods.map(x => ts.factory.createStringLiteral(x.name))))
+        ];
+        if (meta.templateProperties?.length) {
+            properties.push(ts.factory.createPropertyAssignment('templateProps', ts.factory.createArrayLiteralExpression(meta.templateProperties.map(x => ts.factory.createStringLiteral(x)))));
+        }
+        if (meta.numericProperties?.length) {
+            properties.push(ts.factory.createPropertyAssignment('numericProps', ts.factory.createArrayLiteralExpression(meta.numericProperties.map(x => ts.factory.createStringLiteral(x)))));
+        }
+        if (meta.booleanProperties?.length) {
+            properties.push(ts.factory.createPropertyAssignment('boolProps', ts.factory.createArrayLiteralExpression(meta.booleanProperties.map(x => ts.factory.createStringLiteral(x)))));
+        }
+        if (meta.provideAs) {
+            properties.push(ts.factory.createPropertyAssignment('provideAs', ts.factory.createIdentifier(meta.provideAs.symbol.name)));
+        }
+
+        return ts.factory.createObjectLiteralExpression(properties);
+    }
+
+    run(config: Map<ts.InterfaceType, ComponentMetadata>) {
+        const outFile = path.join(path.dirname(this.#configPath), this.out);
+        const placeholder = ts.createSourceFile(outFile, '', ts.ScriptTarget.Latest);
+
+        // REVIEW: We the current approach this seems unnecessary
```

Comment:
Not sure I get it, the config can add files that are not part of the original, thus not imported. Those are now hardcode in the template, but it seems cumbersome to add those instead of letting the analyzer discover them. So this could very well still be needed I think